internal/discovery/discovery_test.go

```diff
@@ -14,6 +14,14 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func normPath(p string) string {
```

Comment:
this looks identical in logic to `resolvePath`, I get since that's an internal function we can't use it here, but could we name it the same and make it a copy paste (the logic is structured slightly differently) to make the identical-ness more obvious?